packages/openapi-v3/src/enhancers/extensions/consolidate.spec.extension.ts

```diff
@@ -0,0 +1,130 @@
+import {bind} from '@loopback/core';
+import compare from 'json-schema-compare';
+import _ from 'lodash';
+import {
+  ISpecificationExtension,
+  isSchemaObject,
+  OpenApiSpec,
+  ReferenceObject,
+  SchemaObject,
+} from '../../types';
+import {asSpecEnhancer, OASEnhancer} from '../types';
+
+/**
+ * A spec enhancer to consolidate OpenAPI specs
+ *
```

Comment:
Could you add more description about the rules? So that users can understand this awesome feature quickly by reading the API doc :) Here is my draft summary(IIUC):

This enhancer consolidates schemas used in multiple places into `/components/schemas` and replaces them with $ref at usage points.

For example, if a schema called  `user` exists in more than one path, it will be consolidated into `/components/schemas/user` and the path objects reference to it as `schema: { $ref: '#/components/schemas/user'}`.

When comparing two schemas, the `description` field is ignored.